mill-db-repo/src/test/java/org/duracloud/mill/manifest/jpa/ManifestStoreIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+        String contentSize2 = "contentSize2";
+        long ms999ago = oneSecondAgo + 1;
+
+        assertTrue(store.addUpdate(account, storeId, spaceId, contentId, contentChecksum2, contentMimetype2,
+                contentSize2,
+                new Date(ms999ago)));
+
+        // get it
+        ManifestItem item2 = store.getItem(account, storeId, spaceId, contentId);
+
+        // verify time modified exactly matches new timestamp
+        assertEquals(ms999ago, item2.getModified().getTime());
+        // verify updated fields
+        assertEquals(contentChecksum2, item2.getContentChecksum());
+        assertEquals(contentMimetype2, item2.getContentMimetype());
+        assertEquals(contentSize2, item2.getContentSize());
+
+        // verify version autoincrement
+        assertEquals(item2.getVersion(), item1.getVersion() + 1);
+    }
+
+    @Test
+    public void testIgnoreUpdateDueToOutOfOrderMessage() throws Exception {
+
+        long currentTime = System.currentTimeMillis();
+        Date timestamp = new Date(currentTime);
+        ManifestItem item1 = addAndRetrieveItem(timestamp);
+        ManifestStore store = getStore();
+
+        // change the checksum
+        long pastTime = currentTime - 1;
+        String newChecksum = "checksum-2";
+        assertFalse(store.addUpdate(account, storeId, spaceId, contentId, newChecksum, contentMimetype,
+                contentSize,
+                new Date(pastTime)));
+
+        // get it
+        ManifestItem item2 = store.getItem(account, storeId, spaceId, contentId);
+
+        // verify time modified exactly matches new timestamp
+        assertEquals(currentTime, item2.getModified().getTime());
+        // verify updated fields
+        assertEquals(contentChecksum, item2.getContentChecksum());
+        // verify version autoincrement
+        assertEquals(item2.getVersion(), item1.getVersion());
+
+    }
+
+    @Test
+    public void testFlagAsDeleted() throws Exception {
+        long currentTime = System.currentTimeMillis();
+        Date past = new Date(currentTime - 1);
+        Date present = new Date(currentTime);
+        ManifestItem item1 = addAndRetrieveItem(present);
+        ManifestStore store = getStore();
+        assertTrue(store.flagAsDeleted(account, storeId, spaceId, contentId, present));
+        store.flagAsDeleted(account, storeId, spaceId, contentId, new Date(currentTime + 1));
+        ManifestItem item2 = store.getItem(account, storeId, spaceId, contentId);
+        assertTrue(item2.isDeleted());
+    }
+
+    @Test
+    public void testFlagAsNotFound() throws Exception {
+        ManifestStore store = getStore();
+        Date present = new Date();
+        assertTrue(store.flagAsDeleted(account, storeId, spaceId, contentId, present));
+        ManifestItem item1 = store.getItem(account, storeId, spaceId, contentId);
+        assertTrue(item1.isDeleted());
+    }
+
+    @Test(expected = NotFoundException.class)
```

Comment:
This is cool, I haven't seen the test annotation used with the expected exception before. Neat.